relayer/src/message_relayer/common/gear/block_listener.rs

```diff
@@ -157,32 +157,76 @@ impl BlockListener {
             }
         }
 
-        let mut finalized_blocks = gear_api.api.subscribe_finalized_blocks().await?;
-        loop {
-            match finalized_blocks.next().await {
-                Some(Err(err)) => {
-                    log::error!("Error receiving finalized block: {err}");
-                    break Err(err);
+        let mut last_finalized_block_number = None;
+        let mut sub = gear_api.subscribe_grandpa_justifications().await?;
+
+        while let Some(justification) = sub.next().await {
+            let justification = justification?;
+
+            let block_hash = justification.commit.target_hash;
+            let block_number = match gear_api.block_hash_to_number(block_hash).await {
```

Comment:
we haven't to make another request to node. The corresponding block number is contained by the justification struct